eslint.precommit.config.mjs

```diff
@@ -0,0 +1,15 @@
+// Pre-commit ESLint config with stricter rules
+import baseConfig from './eslint.config.mjs';
+
+export default [
+  ...baseConfig,
+  {
+    rules: {
+      // Override rules to be stricter for pre-commit
+      'react-hooks/exhaustive-deps': 'error',
```

Comment:
This is the meat of the change, making `exhaustive-deps` an error vs warn.  This way developers will not be able to commit if their dependency array is in violation.